packages/connection-storage/src/connection-storage.ts

```diff
@@ -224,26 +193,18 @@ export class ConnectionStorage {
 
       // While testing, we don't use keychain to store secrets
       if (process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE === 'true') {
-        await fs.writeFile(
-          this.getFilePath(connectionInfo.id),
-          JSON.stringify({ connectionInfo, _id: connectionInfo.id }, null, 2),
-          'utf-8'
-        );
+        await this.userData.write(this.getFileName(connectionInfo.id), {
+          connectionInfo,
+          _id: connectionInfo.id,
```

Comment:
to keep allowing downgrading we may need to do something about this: https://github.com/mongodb-js/compass/blob/abe03298797ce60e25f1c130d6fbdc47d1e4c38f/packages/connection-storage/src/connection-storage.ts#L78

Now that we save that date object with EJSON that line become `new Date({$date: '...'})` which is not a valid date.

It would be nice to add tests to make sure this migration to EJSON is not going to cause issues in both directions (for those who upgrade and those who downgrades), if we don't feel comfortable doing this changes all at once, we can stick to JSON too

Reply:
Yeah, totally worth adding some more thorough testing this time. While setting up a full e2e test for this is way too hard, I think we definitely should add some backwards compat tests, let's grab some files from the current version in the app, store it in fixtures and test against it? Would it be a good way to cover it?

Reply:
Thanks both your feedback. @gribnoysup I had a chat with @mcasimir regarding this change and I'll create a follow up to use EJSON by default. And for now, we will keep using the default (JSON) for most use cases and EJSON is query storage.